src/internal_events/http_client.rs

```diff
@@ -109,3 +109,17 @@ impl<'a, B: HttpBody> std::fmt::Display for FormatBody<'a, B> {
         }
     }
 }
+
+#[derive(Debug)]
+pub struct HttpBodyDataReceived<'a> {
+    pub buf: &'a Vec<u8>,
+}
+
+impl<'a> InternalEvent for HttpBodyDataReceived<'a> {
+    fn emit(self) {
+        // log the buffer content of the configured limit
+        // TODO: make this configurable
+        // TODO: see if changing http.rs to use data_buf.take().unwrap() will cause the below line to be simpler
+        debug!(message = "HttpBodyDataReceived", buffer = format!("{:?}", self.buf.into_iter().take(20).map(|b| *b).collect::<Vec<u8>>()));
```

Comment:
Note the map(|b| *b) was what I noticed would allow the compiler to stop complaining about me using a reference vector